lib/auth/accesspoint/accesspoint.go

```diff
@@ -83,9 +78,97 @@ func (c *AccessCacheConfig) CheckAndSetDefaults() error {
 	return nil
 }
 
-// NewAccessCache builds a cache.Cache instance for a teleport service. This logic has been
-// broken out of lib/service in order to support easier unit testing of process components.
-func NewAccessCache(cfg AccessCacheConfig) (*cache.Cache, error) {
+// NewCacheForServices creates a new cache for a Teleport service that
+// uses the provided services to populate its resource collections.
+func NewCacheForServices(
```

Comment:
Should the methods here be NewCache<i>From</i>Services / NewCache<i>From</i>Client ?

Reply:
Works for me.

Reply:
[340b808](https://github.com/gravitational/teleport/pull/45058/commits/340b808630be3b04c8a8190d5c0320ce7003c23a)